src/vs/workbench/parts/files/browser/fileActions.ts

```diff
@@ -2049,6 +2051,57 @@ export class CompareWithSavedAction extends Action {
 	}
 }
 
+export class CompareWithClipboardAction extends Action {
+
+	public static ID = 'workbench.files.action.compareWithClipboard';
+	public static LABEL = nls.localize('compareWithClipboard', "Compare Active File with Clipboard");
+
+	private static SCHEME = 'compareWithClipboard';
+
+	private toDispose: IDisposable[];
+
+	constructor(
+		id: string,
+		label: string,
+		@IWorkbenchEditorService private editorService: IWorkbenchEditorService,
+		@IInstantiationService instantiationService: IInstantiationService,
+		@ITextModelService textModelService: ITextModelService,
+	) {
+		super(id, label);
+
+		this.enabled = true;
+		this.toDispose = [];
+
+		const provider = instantiationService.createInstance(ClipboardContentProvider);
+		// this.toDispose.push(provider);
+
+		const registrationDisposal = textModelService.registerTextModelContentProvider(CompareWithClipboardAction.SCHEME, provider);
```

Comment:
Can we move the registration into the run method and dispose it after the run has finished? This makes sure we are not keeping the clipboard content provider around for any longer than for the invocation of the action.